theano/tensor/tests/test_basic.py

```diff
@@ -1937,14 +1965,89 @@ def test_py_c_match():
 J1Tester = makeBroadcastTester(
     op=tensor.j1,
     expected=expected_j1,
-    good=_good_broadcast_unary_j,
+    good=_good_broadcast_unary_bessel,
+    grad=_grad_broadcast_unary_bessel,
     eps=2e-10,
     mode=mode_no_scipy,
     skip=skip_scipy)
+
 J1InplaceTester = makeBroadcastTester(
     op=inplace.j1_inplace,
     expected=expected_j1,
-    good=_good_broadcast_unary_j,
+    good=_good_broadcast_unary_bessel,
+    grad=_grad_broadcast_unary_bessel,
+    eps=2e-10,
+    mode=mode_no_scipy,
+    inplace=True,
+    skip=skip_scipy)
+
+JvTester = makeBroadcastTester(
+    op=tensor.jv,
+    expected=expected_jv,
+    good=_good_broadcast_binary_bessel,
```

Comment:
I wasn't sure how to implement the test for the gradient such that only the gradient with respect to the second input is tested (as the gradient with respect to the first input does not have a closed form solution and there isn't a numerical implementation available in `scipy.special` I left it not implemented).

I read through [the guide in the documentation](http://deeplearning.net/software/theano/extending/unittest.html#validating-the-gradient) and tried wrapping the op call in a function which fixes the first input but then couldn't figure out how to integrate this in to the `makeBroadcastTester` interface. I could add separate `test_verify_*_grad` functions to test the gradient separately if it is acceptable to have plain functions as test cases rather than instances of `unittest.TestCase`?